activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -575,6 +599,24 @@ def num_waiting_in_queue # :nodoc:
         @available.num_waiting
       end
 
+      # Discard excess connections from the pool. After a busy period, the
+      # pool may contain more unused connections than is sensible to retain.
+      def reap_excess_connections
+        synchronize do
+          goal = @high_water_mark.value + @spare_count
+          goal = @minimum_size if goal < @minimum_size
+
+          while @connections.size > goal
+            conn = @available.poll or break
+            remove conn
+            @active_connections.increment
```

Comment:
We definitively should have a comment explaining why this is an `increment` (due to `ConnectionLeasingQueue#internal_poll` and `ConnPool#remove` interplay) instead of a `decrement`, otherwise on the cursory reading that looks completely wrong :smile:.